### PR TITLE
backend as a context manager

### LINKED ISSUE #117: exc when cleaning tmpdir
```
Exception ignored in: <bound method Backend.__del__ of <conu.backend.docker.backend.DockerBackend object at 0x7f45d3e30518>>
Traceback (most recent call last):
  File "conu/apidefs/backend.py", line 93, in __del__
TypeError: 'NoneType' object is not callable
```

### PR DESCRIPTION
**Pull request by TomasTomecek**
Fixes #117

### CODE REVIEW

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/0sls](https://da.gd/0sls)__
* Comment: __See details for logs__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/VVxz](https://da.gd/VVxz)__
* Comment: __See details for logs__

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/7pF8](https://da.gd/7pF8)__
* Comment: __See details for logs__

**Comment by TomasTomecek** [comment]
C'mon CI, you can do this!

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: failure, return code = __1__ 
* Result url: __[https://da.gd/5DfoP](https://da.gd/5DfoP)__
* Comment: __See details for logs__

**Comment by dhodovsk** [comment]
This CI problem seems to be different from what we've seen before. Now containers see each other, but expected_output has one extra newline
```
-------stdout-------
b'Password: \n ?column? \n----------\n        1\n(1 row)\n'
-------expected-------
b'Password: \n ?column? \n----------\n        1\n(1 row)\n\n'
```

File: conu/apidefs/backend.py [hunk 1]
```diff
@@ -63,7 +63,16 @@ def set_logging(
...
+    This class groups classes and functionality related to a specific backend.
+
+    We strongly advise you to use backend as a context manager:
+
+    ::
+
+        with SomeBackend() as backend:
+            backend...
+
+    This will ensure that temporary directory associated with the backend instance will be cleaned.
```
Comment (dhodovsk) [inline conu/apidefs/backend.py]: Reading this I was a bit confused what directory is this referring to. There might be also a little description of it and how it is used.

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: success, return code = __0__ 
* Result url: __[https://da.gd/9ASJY](https://da.gd/9ASJY)__
* Comment: __See details for logs__

**Comment by TomasTomecek** [comment]
updated, maybe even overkilled

**Comment by dhodovsk** [comment]
No, its great now! :)

**Comment by psklenar** [comment]
## jenkins/superCI comment
* Status: success, return code = __0__ 
* Result url: __[https://da.gd/KRw5](https://da.gd/KRw5)__
* Comment: __See details for logs__

**Pull request merged**